e2e test is broken for Mac OS X

**OS**: Mac OS X Sierra 10.12.3
**Steps**: `make test`
**Error:**: 
```
Running tests:
ok  	k8s.io/git-sync/cmd/git-sync	0.046s
?   	k8s.io/git-sync/pkg/version	[no test files]

Checking gofmt: PASS

Checking go vet: PASS

building: bin/amd64/git-sync
Sending build context to Docker daemon 8.736 MB
Step 1/10 : FROM alpine:3.4
 ---> 0766572b4bac
Step 2/10 : MAINTAINER Tim Hockin <thockin@google.com>
 ---> Using cache
 ---> 0eac37411a22
Step 3/10 : ADD bin/amd64/git-sync /git-sync
 ---> Using cache
 ---> 65a54a0c5443
Step 4/10 : ENV GIT_SYNC_DEST /git
 ---> Using cache
 ---> 6bade5a07d01
Step 5/10 : RUN apk update --no-cache && apk add     ca-certificates     coreutils     git     openssh-client
 ---> Using cache
 ---> 8cb820470a33
Step 6/10 : RUN mv /usr/bin/ssh /usr/bin/ssh-binary
 ---> Using cache
 ---> c55cce37bd39
Step 7/10 : ADD ssh-wrapper.sh /usr/bin/ssh
 ---> Using cache
 ---> 1a61fe46952d
Step 8/10 : RUN chmod 755 /usr/bin/ssh
 ---> Using cache
 ---> 029ee069f00d
Step 9/10 : USER nobody:nobody
 ---> Using cache
 ---> cca0e247c918
Step 10/10 : ENTRYPOINT /git-sync
 ---> Using cache
 ---> 71aaf60e7e1b
Successfully built 71aaf60e7e1b
container: e2e/git-sync-amd64:v2.0.4-2-g5e2d975
test root is /tmp/git-sync-test.17388
Initialized empty Git repository in /private/tmp/git-sync-test.17388/repo/.git/
testcase head-once: PASS
testcase default-sync: FAIL:  /tmp/git-sync-test.17388/root/link does not exist or is not a symlink
ytussupbekov      2457   0.1  0.1 556625368  10996 s000  S+   12:21AM   0:00.02 docker run -i -u 2064180707:113584762 -v /tmp/git-sync-test.17388:/tmp/git-sync-test.17388 e2e/git-sync-amd64:v2.0.4-2-g5e2d975 --logtostderr --v=5 --wait=0.1 --repo=/tmp/git-sync-test.17388/repo --root=/tmp/git-sync-test.17388/root --dest=link
ytussupbekov      2461   0.0  0.0  2442020    808 s000  S+   12:21AM   0:00.00 grep git-sync
make: *** [test] Error 1
```

**Cause**: The reason being `/tmp` directory itself in Mac OS X is a symlink to `/private/tmp`. Suggested fix could be to identify architecture `darwin`-> and run tests in `/private/tmp`. For other architecture, keep it as it is. Though I am not entirely sure on the state of `test_e2e.sh` and usefulness of having the test running in Mac OS X. Thoughts?

--- a/test_e2e.sh
+++ b/test_e2e.sh
@@ -13,7 +13,7 @@ function testcase() {
 
 function fail() {
     echo "FAIL: " "$@"
-    sleep 1
+    sleep 2
     pkill git-sync || true
     ps auxw | grep git-sync
     exit 1
@@ -26,17 +26,18 @@ function pass() {
 }
 
 function assert_link_exists() {
-    if [[ -L "$1" ]]; then
-        return
+    if ! [[ -e "$1" ]]; then
+        fail "$1 does not exist"
     fi
-    fail "$1 does not exist or is not a symlink"
+    if ! [[ -L "$1" ]]; then
+        fail "$1 is not a symlink"
+    fi
 }
 
 function assert_file_exists() {
-    if [[ -f "$1" ]]; then
-        return
+    if ! [[ -f "$1" ]]; then
+        fail "$1 does not exist"
     fi
-    fail "$1 does not exist"
 }
 
 function assert_file_eq() {
@@ -123,20 +124,20 @@ GIT_SYNC \
     --repo="$REPO" \
     --root="$ROOT" \
     --dest="link" > "$DIR"/log."$TESTCASE" 2>&1 &
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
 # Move forward
 echo "$TESTCASE 2" > "$REPO"/file
 git -C "$REPO" commit -qam "$TESTCASE 2"
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 2"
 # Move backward
 git -C "$REPO" reset -q --hard HEAD^
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
@@ -159,20 +160,20 @@ GIT_SYNC \
     --rev=HEAD \
     --root="$ROOT" \
     --dest="link" > "$DIR"/log."$TESTCASE" 2>&1 &
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
 # Move HEAD forward
 echo "$TESTCASE 2" > "$REPO"/file
 git -C "$REPO" commit -qam "$TESTCASE 2"
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 2"
 # Move HEAD backward
 git -C "$REPO" reset -q --hard HEAD^
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
@@ -197,7 +198,7 @@ GIT_SYNC \
     --branch="$BRANCH" \
     --root="$ROOT" \
     --dest="link" > "$DIR"/log."$TESTCASE" 2>&1 &
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
@@ -206,7 +207,7 @@ git -C "$REPO" checkout -q "$BRANCH"
 echo "$TESTCASE 2" > "$REPO"/file
 git -C "$REPO" commit -qam "$TESTCASE 2"
 git -C "$REPO" checkout -q master
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 2"
@@ -214,7 +215,7 @@ assert_file_eq "$ROOT"/link/file "$TESTCASE 2"
 git -C "$REPO" checkout -q "$BRANCH"
 git -C "$REPO" reset -q --hard HEAD^
 git -C "$REPO" checkout -q master
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
@@ -238,7 +239,7 @@ GIT_SYNC \
     --rev="$TAG" \
     --root="$ROOT" \
     --dest="link" > "$DIR"/log."$TESTCASE" 2>&1 &
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
@@ -246,21 +247,21 @@ assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
 echo "$TESTCASE 2" > "$REPO"/file
 git -C "$REPO" commit -qam "$TESTCASE 2"
 git -C "$REPO" tag -af "$TAG" -m "$TESTCASE 2" >/dev/null
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 2"
 # Move the tag backward
 git -C "$REPO" reset -q --hard HEAD^
 git -C "$REPO" tag -af "$TAG" -m "$TESTCASE 3" >/dev/null
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
 # Add something after the tag
 echo "$TESTCASE 3" > "$REPO"/file
 git -C "$REPO" commit -qam "$TESTCASE 3"
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
@@ -287,7 +288,7 @@ GIT_SYNC \
     --rev="$TAG" \
     --root="$ROOT" \
     --dest="link" > "$DIR"/log."$TESTCASE" 2>&1 &
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
@@ -297,7 +298,7 @@ echo "$TESTCASE 2" > "$REPO"/file
 git -C "$REPO" commit -qam "$TESTCASE 2"
 git -C "$REPO" tag -af "$TAG" -m "$TESTCASE 1" >/dev/null
 git -C "$REPO" checkout -q master
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 2"
@@ -306,7 +307,7 @@ git -C "$REPO" checkout -q "$BRANCH"
 git -C "$REPO" reset -q --hard HEAD^
 git -C "$REPO" tag -af "$TAG" -m "$TESTCASE 1" >/dev/null
 git -C "$REPO" checkout -q master
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
@@ -315,7 +316,7 @@ git -C "$REPO" checkout -q "$BRANCH"
 echo "$TESTCASE 3" > "$REPO"/file
 git -C "$REPO" commit -qam "$TESTCASE 3"
 git -C "$REPO" checkout -q master
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
@@ -323,7 +324,7 @@ assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
 git -C "$REPO" checkout -q "$BRANCH"
 git -C "$REPO" tag -af "$TAG" -m "$TESTCASE 3" >/dev/null
 git -C "$REPO" checkout -q master
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 3"
@@ -346,20 +347,20 @@ GIT_SYNC \
     --rev="$REV" \
     --root="$ROOT" \
     --dest="link" > "$DIR"/log."$TESTCASE" 2>&1 &
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
 # Commit something new
 echo "$TESTCASE 2" > "$REPO"/file
 git -C "$REPO" commit -qam "$TESTCASE 2"
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
 # Revert the last change
 git -C "$REPO" reset -q --hard HEAD^
-sleep 1
+sleep 2
 assert_link_exists "$ROOT"/link
 assert_file_exists "$ROOT"/link/file
 assert_file_eq "$ROOT"/link/file "$TESTCASE 1"

